docs/static/performance-checklist.asciidoc

```diff
@@ -49,59 +50,132 @@ sure-fire way to create a confusing situation.
+A pipeline's `queue_backpressure` flow metric reflects how much time the inputs are spending attempting to push events into the queue.
+The metric isn't precisely comparable across pipelines, but instead allows you to compare a single pipeline's current behaviour to _itself_ over time.
+When this metric is growing, we generally need to look _downstream_ at the pipeline's filters and outputs to see if they are using resources effectively, have sufficient resources allocated, or are experiencing back-pressure of their own.
+
+NOTE: A persisted queue offers durability guarantees and can absorb back-pressure for longer than the default in-memory queue, but once it is full it too propagates back-pressure.
+The `queue_persisted_growth_events` flow metric is useful measure of how much back-pressure is being actively absorbed by the persisted queue, and should trend toward zero (or less) over the pipeline's lifetime.
+
+[float]
+[[tuning-logstash-settings]]
+==== Tuning-related settings
+
+The Logstash defaults are chosen to provide fast, safe performance for most users.
+However if you notice performance issues, you may need to modify some of the defaults.
+Logstash provides the following configurable options for tuning pipeline performance: `pipeline.workers`, `pipeline.batch.size`, and `pipeline.batch.delay`.
+
+For more information about setting these options, see <<logstash-settings-file>>.
+
+Make sure you've read the <<performance-troubleshooting>> before modifying these options.
 
-* The `pipeline.batch.delay` setting rarely needs to be tuned. This setting adjusts the latency of the Logstash pipeline. Pipeline batch delay is the maximum amount of time in milliseconds that Logstash waits for new messages after receiving an event in the current pipeline worker thread. After this time elapses, Logstash begins to execute filters and outputs.The maximum time that Logstash waits between receiving an event and processing that event in a filter is the product of the `pipeline.batch.delay` and  `pipeline.batch.size` settings.
+* The `pipeline.workers` setting determines how many threads to run for filter and output processing.
+  If you find that events are backing up, or that the CPU is not saturated, consider increasing the value of this parameter to make better use of available processing power.
+  Good results can even be found increasing this number past the number of available processors as these threads may spend significant time in an I/O wait state when writing to external systems.
+  Legal values for this parameter are positive integers.
```

Comment:
```suggestion
```

I agree -- it was here because of the net-no-change refactor to support the one-sentence-per-line convention, but can be ✂️ here.